aten/src/ATen/native/README.md

```diff
@@ -0,0 +1,109 @@
+ATen native functions are a mechanism to write ATen methods which only
+make use of other ATen operations (e.g., it is not necessary to bind into
+TH/THC code).  These functions are declared in this file and then folded
+into the ATen code generation process.
+
+## Registering it in `native_functions.yaml`
+
+The first step is to write an entry for your function in
+`native_functions.yaml`.  The format is as follows:
+
+```
+- func: func_name(ArgType arg0[=default], ArgType arg1[=default], ...) -> ReturnType
+```
+
+ArgType(s) are allowed to be simple types understood by ATen
+(e.g. `Tensor`, `TensorList`, `IntList`, `int64_t`, `double`, ...).
+`Tensor?` means that the tensor is optional; if it is not passed by
```

Comment:
These terms aren't well defined.  What is optional?  What does not passed by the user mean?  What does checking if it is the case mean?  What API is this talking about?